single-line-diagram-core/src/main/java/com/powsybl/sld/svg/DefaultDiagramStyleProvider.java

```diff
@@ -221,49 +213,39 @@ public String getIdWireStyle(Edge edge) {
         return attributes;
     }
 
-    private Optional<String> getColor3WT(String transformerId, String nameSubComponent, boolean rotateSVG, String vlId) {
-        Optional<String> color = Optional.empty();
-
-        ThreeWindingsTransformer transformer = network.getThreeWindingsTransformer(transformerId);
-        ThreeWindingsTransformer.Side otherSide = ThreeWindingsTransformer.Side.ONE;
-
-        VoltageLevel leg1Vl = transformer.getLeg1().getTerminal().getVoltageLevel();
-        VoltageLevel leg2Vl = transformer.getLeg2().getTerminal().getVoltageLevel();
-        VoltageLevel leg3Vl = transformer.getLeg3().getTerminal().getVoltageLevel();
+    private Optional<String> getColor3WT(Fictitious3WTNode node, String nameSubComponent, boolean rotateSVG, String vlId) {
+        Map<Feeder3WTNode.Side, String> idsLegs = node.getIdsLegs();
+        Map<Feeder3WTNode.Side, Double> vNomsLegs = node.getvNomsLegs();
 
-        ThreeWindingsTransformer.Side side1 = ThreeWindingsTransformer.Side.ONE;
-        ThreeWindingsTransformer.Side side2 = ThreeWindingsTransformer.Side.TWO;
-        ThreeWindingsTransformer.Side side3 = ThreeWindingsTransformer.Side.THREE;
+        Feeder3WTNode.Side otherSide = Feeder3WTNode.Side.ONE;
 
         if (nameSubComponent.equals(WINDING1)) {
-            if (leg1Vl.getId().equals(vlId)) {
-                otherSide = !rotateSVG ? side3 : side2;
-            } else if (leg2Vl.getId().equals(vlId)) {
-                otherSide = !rotateSVG ? side3 : side1;
-            } else if (leg3Vl.getId().equals(vlId)) {
-                otherSide = !rotateSVG ? side2 : side1;
+            if (idsLegs.get(Feeder3WTNode.Side.ONE).equals(vlId)) {
+                otherSide = !rotateSVG ? Feeder3WTNode.Side.THREE : Feeder3WTNode.Side.TWO;
```

Comment:
Is it the right place to deal with rotation? This class should only be responsible for style but not for layout.

Reply:
In another issue, I'm going to try to move the svg component rotation outside the default diagram style provider methods (in fact, in the SVG writer before calling the style provider getNodeSVGStyle method), and just use this information for generating the right style ...